solutions/usaco-569.mdx

```diff
@@ -2,106 +2,313 @@
 id: usaco-569
 source: USACO Bronze 2015 December
 title: Contaminated Milk
-author: Jesse Choe
+author: Jesse Choe, Kevin Sheng
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_badmilk_bronze_dec15.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_badmilk_bronze_dec15.html)
+
+## Explanation
+
+We can do a simple brute force to find out which milk could possibly be bad,
+and then see how many people that milk could have infected in total.
+
+Since FJ needs one medicine for each person who could possibly get sick,
+we can just take the maximum possible number of infected people over all
+milk types that could have possibly gone bad.
+
+The main issue is checking if a milk could have been the one that went bad.
+To resolve this, let's treat both a person drinking milk and a person getting sick
+as a type of *event*.
+Then, we can have a list of events sorted by time. The sample case in this format
+would be as follows:
+
+<center>
+
+| Time | Person | Drink |
+|:----:|:------:|:-----:|
+|1|1|1|
+|1|1|4|
+|2|1|2|
+|3|3|1|
+|3|1|NA|
+|4|1|3|
+|5|2|1|
+|7|2|2|
+|8|2|NA|
+
+</center>
+
+With this list of events, we can then go through marking the
+people that could have possibly gotten sick and checking each
+event where a person got sick if they could have possibly gotten sick.
+
+## Implementation
+
+**Time Complexity: ** $\mathcal{O}(N^2)$
```

Comment:
huh isn't this more like M * (D + S) or something?

Reply:
oh yeah i just bs'ed a complexity i'll change it